core/src/main/scala/org/http4s/multipart/Multipart.scala

```diff
@@ -0,0 +1,48 @@
+package org.http4s
+package multipart
+
+import scala.util.Random
+
+import org.http4s._
+import org.http4s.EntityEncoder._
+import org.http4s.headers.{ `Content-Type` ⇒ ContentType, `Content-Disposition` => ContentDisposition }
+import org.http4s.MediaType._
+
+import scalaz._
+import Scalaz._
+
+import scodec.bits.ByteVector
+
+final case class Name(value:String) extends AnyVal
+sealed trait Part
+final case class FormData(name:Name,
+                          content: Entity,
+                          contentType: Option[ContentType] = None) extends Part
+
+final case class Multipart(val parts: Seq[Part],val boundary:Boundary = Boundary.create) {
```

Comment:
If the intent is to eventually parse the `Part`s in a streaming fashion, should `parts` here be made into a `Process[Task, Part]` instead of a `Seq[Part]` for future compatibility?